keyring/backends/macOS/__init__.py

```diff
@@ -32,36 +33,54 @@ def priority(cls):
             raise RuntimeError("Security API unavailable")
         return 5
 
-    def set_password(self, service, username, password):
+    def set_password(self, service, username, password, keychain=""):
```

Comment:
Using `""` is a bad default value... unless the empty string is actually meaningful as a default value.